wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/jpa/JpaStackDao.java

```diff
@@ -0,0 +1,138 @@
+    public void create(StackImpl stack) throws ConflictException, ServerException {
+        requireNonNull(stack, "Required non-null stack");
+        try {
+            doCreate(stack);
+        } catch (DuplicateKeyException x) {
+            throw new ConflictException(format("Stack with id '%s' or name '%s' already exists", stack.getId(), stack.getName()));
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Override
+    @Transactional
+    public StackImpl getById(String id) throws NotFoundException, ServerException {
+        requireNonNull(id, "Required non-null id");
+        try {
+            final StackImpl stack = managerProvider.get().find(StackImpl.class, id);
+            if (stack == null) {
+                throw new NotFoundException(format("Stack with id '%s' doesn't exist", id));
+            }
+            return stack;
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Override
+    public void remove(String id) throws ServerException {
+        requireNonNull(id, "Required non-null id");
+        try {
+            doRemove(id);
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Override
+    public StackImpl update(StackImpl update) throws NotFoundException, ServerException, ConflictException {
+        requireNonNull(update, "Required non-null update");
+        try {
+            return doUpdate(update);
+        } catch (DuplicateKeyException x) {
+            throw new ConflictException("Stack with name '%s' already exists");
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<StackImpl> searchStacks(@Nullable String user,
+                                        @Nullable List<String> tags,
+                                        int skipCount,
+                                        int maxItems) throws ServerException {
+        if (tags == null) {
+            tags = emptyList();
+        }
+        try {
+            return managerProvider.get()
+                                  .createNamedQuery("Stack.search", StackImpl.class)
+                                  .setParameter("user", user)
+                                  .setParameter("tags", tags)
+                                  .setParameter("tagsSize", tags.size())
+                                  .setMaxResults(maxItems)
+                                  .setFirstResult(skipCount)
+                                  .getResultList();
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Transactional
+    protected void doCreate(StackImpl stack) {
+        managerProvider.get().persist(stack);
```

Comment:
`@Transactional` intercepts action with something like 
```
tx.begin();
action();
tx.commit();
```
from the context `tx.commit()` throws `DuplicateKeyException` and `RollbackException `hierarchies which we need to catch and rethrow as `ConflcitException` and so on. If i put `@Transactional` directly on method it wouldn't be possible to catch errors.